docs/spec/spec.md

```diff
@@ -0,0 +1,694 @@
+# Specification: Ballerina Kafka Library
+
+_Owners_: @shafreenAnfar @dilanSachi @aashikam    
+_Reviewers_: @shafreenAnfar @aashikam  
+_Created_: 2020/10/28   
+_Updated_: 2021/12/07  
+_Issue_: [#2186](https://github.com/ballerina-platform/ballerina-standard-library/issues/2186)  
+
+# Introduction
+This is the specification for Kafka standard library which is used to send and receive messages by connecting to a Kafka server.
+This library is programmed in the [Ballerina programming language](https://ballerina.io/), which is an open-source programming language for the cloud
+that makes it easier to use, combine, and create network services.
+
+# Contents
+
+1. [Overview](#1-overview)  
+2. [Configurations](#2-configurations)  
+   2.1 [Producer Configurations](#21-producer-configurations)   
+   2.2 [Consumer/Listener Configurations](#22-consumerlistener-configurations)    
+   2.3 [Security Configurations](#23-security-configurations)      
+   2.4 [Common Configurations](#24-common-configurations)   
+   *   2.4.1 [TopicPartition](#241-topicpartition)     
+   *   2.4.2 [PartitionOffset](#242-partitionoffset)   
+3. [Producer](#3-producer)  
+   3.1 [Connection](#31-connection)     
+   *    3.1.1 [Insecure Connection](#311-insecure-connection)   
+   *    3.1.2 [Secure Connection](#312-secure-connection)   
+   3.2 [Producer Usage](#32-producer-usage)     
+4. [Consumer](#4-consumer)  
```

Comment:
I think under consumer we can have two sections as `Client` and `Listener`

Reply:
Updated